<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>ROSeq</title>

<style type="text/css">
a.anchor-section {margin-left: 10px; visibility: hidden; color: inherit;}
a.anchor-section::before {content: '#';}
.hasAnchor:hover a.anchor-section {visibility: visible;}
</style>
<script>// Anchor sections v1.0 written by Atsushi Yasumoto on Oct 3rd, 2020.
document.addEventListener('DOMContentLoaded', function() {
  // Do nothing if AnchorJS is used
  if (typeof window.anchors === 'object' && anchors.hasOwnProperty('hasAnchorJSLink')) {
    return;
  }

  const h = document.querySelectorAll('h1, h2, h3, h4, h5, h6');

  // Do nothing if sections are already anchored
  if (Array.from(h).some(x => x.classList.contains('hasAnchor'))) {
    return null;
  }

  // Use section id when pandoc runs with --section-divs
  const section_id = function(x) {
    return ((x.classList.contains('section') || (x.tagName === 'SECTION'))
            ? x.id : '');
  };

  // Add anchors
  h.forEach(function(x) {
    const id = x.id || section_id(x.parentElement);
    if (id === '') {
      return null;
    }
    let anchor = document.createElement('a');
    anchor.href = '#' + id;
    anchor.classList = ['anchor-section'];
    x.classList.add('hasAnchor');
    x.appendChild(anchor);
  });
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">ROSeq</h1>



<div id="roseq" class="section level1">
<h1>ROSeq</h1>
<p>Modeling expression ranks for noise-tolerant differential expression analysis of scRNA-Seq data</p>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>ROSeq - A rank based approach to modeling gene expression with filtered and normalized read count matrix. ROSeq takes filtered and normalized read matrix and cell-annotation/condition as input and determines the differentially expressed genes between the contrasting groups of single cells. One of the input parameters is the number of cores to be used.</p>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>The developerâ€™s version of the R package can be installed with the following R commands:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="cf">if</span> (<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">    <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co"># The following initializes usage of Bioc devel</span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="dt">version=</span><span class="st">'devel'</span>)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"></a>
<a class="sourceLine" id="cb1-7" data-line-number="7">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;ROSeq&quot;</span>)</a></code></pre></div>
<p>The githubâ€™s version of the R package can be installed with the following R commands:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">install_github</span>(<span class="st">'krishan57gupta/ROSeq'</span>)</a></code></pre></div>
</div>
<div id="vignette-tutorial" class="section level2">
<h2>Vignette tutorial</h2>
<p>This vignette uses the Tung dataset, which is already inbuilt in the package, to demonstrate a standard pipeline.</p>
</div>
<div id="example" class="section level2">
<h2>Example</h2>
<p>Libraries need to be loaded before running.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">library</span>(ROSeq)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw">library</span>(edgeR)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; Loading required package: limma</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">library</span>(limma)</a></code></pre></div>
<div id="loading-tung-dataset" class="section level3">
<h3>Loading tung dataset</h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">samples&lt;-<span class="kw">list</span>()</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">samples<span class="op">$</span>count&lt;-ROSeq<span class="op">::</span>L_Tung_single<span class="op">$</span>NA19098_NA19101_count</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">samples<span class="op">$</span>group&lt;-ROSeq<span class="op">::</span>L_Tung_single<span class="op">$</span>NA19098_NA19101_group</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">samples<span class="op">$</span>count[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt;                 NA19098.r1.A01 NA19098.r1.A02 NA19098.r1.A03 NA19098.r1.A04</span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt; ENSG00000237683              0              0              0              1</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt; ENSG00000187634              0              0              0              0</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; ENSG00000188976              3              6              1              3</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; ENSG00000187961              0              0              0              0</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; ENSG00000187583              0              0              0              0</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt;                 NA19098.r1.A05</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; ENSG00000237683              0</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt; ENSG00000187634              0</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt; ENSG00000188976              4</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt; ENSG00000187961              0</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; ENSG00000187583              0</span></a></code></pre></div>
</div>
<div id="data-preprocessing" class="section level3">
<h3>Data Preprocessing:</h3>
<div id="cells-and-genes-filtering-then-voom-transformation-after-tmm-normalization" class="section level4">
<h4>Cells and genes filtering then voom transformation after TMM normalization</h4>
<p>Below commands can be used for Cell/gene filtering, TMM normalization and voom transformation. The user is free to use an alternative preprocessing strategy while using different filtering/normalization methods.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">gene_names&lt;-<span class="kw">rownames</span>(samples<span class="op">$</span>count)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">samples<span class="op">$</span>count&lt;-<span class="kw">apply</span>(samples<span class="op">$</span>count,<span class="dv">2</span>,<span class="cf">function</span>(x) <span class="kw">as.numeric</span>(x))</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw">rownames</span>(samples<span class="op">$</span>count)&lt;-gene_names</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">samples<span class="op">$</span>count&lt;-samples<span class="op">$</span>count[,<span class="kw">colSums</span>(samples<span class="op">$</span>count<span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2000</span>]</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">gkeep&lt;-<span class="kw">apply</span>(samples<span class="op">$</span>count,<span class="dv">1</span>,<span class="cf">function</span>(x) <span class="kw">sum</span>(x<span class="op">&gt;</span><span class="dv">2</span>)<span class="op">&gt;=</span><span class="dv">3</span>)</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">samples<span class="op">$</span>count&lt;-samples<span class="op">$</span>count[gkeep,]</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">samples<span class="op">$</span>count&lt;-limma<span class="op">::</span><span class="kw">voom</span>(ROSeq<span class="op">::</span><span class="kw">TMMnormalization</span>(samples<span class="op">$</span>count))</a></code></pre></div>
</div>
</div>
<div id="roseq-analysis." class="section level3">
<h3>ROSeq analysis.</h3>
<p>Input: gene expression matrix with genes in rows and cells in columns. Condition/group annotation of cells also need to be supplied. User can set numCores based the hardware specifications in her computer.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">output&lt;-<span class="kw">ROSeq</span>(<span class="dt">countData=</span>samples<span class="op">$</span>count<span class="op">$</span>E, <span class="dt">condition =</span> samples<span class="op">$</span>group, <span class="dt">numCores=</span><span class="dv">1</span>)</a></code></pre></div>
</div>
<div id="showing-results-are-in-the-form-of-pvals-and-padj" class="section level3">
<h3>Showing results are in the form of pVals and pAdj</h3>
<div id="p_vals-p_value-unadjusted" class="section level5">
<h5>p_Vals : p_value (unadjusted)</h5>
</div>
<div id="p_adj-adjusted-p-value-based-on-fdr-method" class="section level5">
<h5>p_Adj : Adjusted p-value, based on FDR method</h5>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">output[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,]</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co">#&gt;                     pVals      pAdj</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; ENSG00000237683 0.6741425 0.9321651</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; ENSG00000188976 0.7484244 0.9426495</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; ENSG00000187608 0.2282451 0.8481636</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; ENSG00000188157 0.5138812 0.9082800</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; ENSG00000131591 0.1235577 0.7438811</span></a></code></pre></div>
</div>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
